Gems/Atom/Tools/AtomToolsFramework/Code/Source/Viewport/RenderViewportWidget.cpp

```diff
@@ -187,10 +186,23 @@ namespace AtomToolsFramework
         return eventHandled && shouldConsumeEvent;
     }
 
-    void RenderViewportWidget::OnTick([[maybe_unused]]float deltaTime, AZ::ScriptTimePoint time)
+    void RenderViewportWidget::OnRenderTick()
     {
+        AZ::ScriptTimePoint time(AZStd::chrono::system_clock::now());
+        m_lastDeltaTime = time.Get() - m_time.Get();
         m_time = time;
-        m_controllerList->UpdateViewport({GetId(), AzFramework::FloatSeconds(deltaTime), m_time});
+        
+        m_controllerList->UpdateViewport({GetId(), m_lastDeltaTime, m_time});
+    }
```

Comment:
If the time delta we're receiving in the OnTick() is different from last-current, doesn't that mean we're updating everything else connected to the tick bus with the same wrong time deltas? Or did we switch to a fixed time step method for the tick bus that I missed?

The fix looks good, but I am just generally concerned about what the rest of the components connected to the tick bus are receiving then. EMFX at least would expect the exact same thing that we do now in the RenderViewportWidget, and everything else will lead to incorrect results as well.

Reply:
+1 - I feel like I don't know enough about this but I'd also like to better understand why standard `OnTick` is wrong and couldn't be fixed, or maybe for this specific issue we should have `OnTick` and `OnRenderTick` (`controllerList` would need two update calls then) and can use whichever is appropriate?